api/task_status.go

```diff
@@ -74,11 +88,33 @@ func (h *taskStatusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		statuses[taskName] = status
 	}
 
+	// if user requested a specific task that does not have events, check if
+	// a driver exists
+	if taskName != "" {
+		if _, ok := data[taskName]; !ok {
+			if d, ok := h.drivers[taskName]; ok {
+				statuses[taskName] = makeTaskStatusUnknown(d.Task())
+			}
+		}
+	}
+
+	// if user requested all tasks and status filter applicable, check driver
+	// for tasks without events
+	if taskName == "" && (filter == "" || filter == StatusUnknown) {
+		for tN, d := range h.drivers {
+			if _, ok := data[tN]; !ok {
+				statuses[tN] = makeTaskStatusUnknown(d.Task())
+			}
+		}
+	}
+
 	jsonResponse(w, http.StatusOK, statuses)
 }
 
-// makeTaskStatus takes event data for a task and returns an overall task status
-func makeTaskStatus(taskName string, events []event.Event, version string) TaskStatus {
+// makeTaskStatus takes event data for a task and returns a task status
+func makeTaskStatus(taskName string, events []event.Event, task driver.Task,
```

Comment:
the `task driver.Task` has the name, and the caller of this function should already ensure that `taskName == driver.Task.Name`, so an option could be to remove `taskName` parameter to condense the function signature.

Reply:
I like that! Thanks for the suggestion.